### PR TITLE
Transform MTK.jl equations into StructuralIdentifiability.jl `ODE` object

### PR DESCRIPTION
**Pull request by iliailmer**
To make the package compatible with MTK equations, this PR adds `PreprocessODE` function.

Example usage:

```julia
using ModelingToolkit, StructuralIdentifiability

ModelingToolkit.@parameters θ1, θ2, θ3, θ4
ModelingToolkit.@variables t, x1, x2, u1, y1

D = Differential(t)
eqs = [D(x1) ~ x1^2 * θ1 + θ2 * x1 * x2 + u1, D(x2) ~ θ3 * x1^2 + θ4 * x1 * x2];
outputs = [y1 ~ x1];

ode = PreprocessODE(eqs, outputs, [x1, x2], [y1], [u1], [θ1, θ2, θ3,θ4], t)

assess_identifiability(ode)
```

To do:
- [ ] Currently does not work with latex-like subscripts, throws string indexing error
- [x] Tests

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
`assess_identifiability` should be a function that acts on the `ODESystem`.

File: src/ODE.jl [hunk 1]
```diff
@@ -333,3 +332,34 @@ function Base.show(io::IO, ode::ODE)
...
 #------------------------------------------------------------------------------
+"""
+    function PreprocessODE(eqs, outputs, x, y, u, θ)
+    
+Input:
+- `eqs` - array of ModelingToolkit differential equations
+- `outputs` - array of output equations
+- `x` - array of state variables
+- `y` - array of output function names
+- `θ` - array of parameter names
```
Comment (pogudingleb) [inline src/ODE.jl:343]: `u` seems to be missing in the spec

**Comment by pogudingleb** [comment]
Hi Ilia,

Thank you for the code!

First, I completely agree with Chris: a user should be able to just run the main functions `assess_identifiability` and `assess_local_identifiability` on the `ODESystem`, and your function should be called internally. By the way, I am still learn the Julia way of doing things, wouldn't be natural to do this as a promotion rule?

I would like to point out a couple of issues:

1. The current implementation does not allow rational fucntions, at least both lines below fail (the first one because of Symbolics, the second because of Nemo)

```
eqs = [D(x1) ~ x1^2 * θ1 + θ2 * x1 // x2 + u1, D(x2) ~ θ3 * x1^2 + θ4 * x1 * x2];
eqs = [D(x1) ~ x1^2 * θ1 + θ2 * x1 / x2 + u1, D(x2) ~ θ3 * x1^2 + θ4 * x1 * x2];
```

2. I have often seen people typing integer constants with a floating point (especially if the model was produced by other code), it would be great if this would work:

```
eqs = [D(x1) ~ x1^2 * θ1 + θ2 * x1 * x2 + 2.0 * u1, D(x2) ~ θ3 * x1^2 + θ4 * x1 * x2];
```

3. In general, I think there should be a check one whether the equations involve special functions, so that this is reported not via a lengthy stacktrace.

Based on this, I would encourage you not to put the tests into a todo list but do them right away including the cases above.

**Comment by iliailmer** [comment]
Two new functions added (with tests): `assess_identifiability` and `assess_local_identifiability`. Both accept `ODESystem`, convert it to `ODE` type object and then run usual algorithms.

Because of the limitation with denominators, they only work for functions without denominators, simple tests.

The other two checks (integer coefficients, special functions) are not done yet.

File: src/ODE.jl [hunk 2]
```diff
@@ -333,3 +332,38 @@ function Base.show(io::IO, ode::ODE)
...
+
+Output: 
+- `ODE` object containing required data for identifiability assessment
+"""
+function PreprocessODE(diff_eqs, out_eqs, state_vars, outputs, inputs, params)# , t)
+    # D = Differential(t)
+    
+	input_symbols = vcat(state_vars, outputs, inputs, params)
+	generators = string.(input_symbols)
+    generators = map(g->replace(g, "(t)"=>""), generators)
```
Comment (pogudingleb) [inline src/ODE.jl:354]: What is with the identation?

**Comment by pogudingleb** [comment]
Thank you, Ilia! Couple more comments:
1. I see that you pass the output separately. Would it be appropriate to use the `observed` field of [`ODESystem`](https://juliahub.com/docs/ModelingToolkit/Qmdqu/3.20.1/systems/ODESystem/)? I am not sure what is exactly its semantics.
2. I believe that it would be good to write a separate function to convert MTK expression into a rational function because you do this in several places: in `PreprocessODE` and in the `assess...` functions. It would be great if rational functions and special functions were treated properly and uniformly in all these places.

**Comment by ChrisRackauckas** [comment]
>I see that you pass the output separately. Would it be appropriate to use the observed field of ODESystem? I am not sure what is exactly its semantics.

Yes.

File: test/mtk_compat.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        D(S) ~ μ - bi * S * I - bw * S * W - μ * S + a * R,
+        D(I) ~ bw * S * W + bi * S * I - (γ + μ) * I,
+        D(W) ~ ξ * (I - W),
+        D(R) ~ γ * I - (μ + a) * R
+    ]
+    de = ODESystem(eqs, t, [S, I, W, R], [μ, bi, bw, a, ξ, γ, k], observed=[y ~ k * I], name=:TestSIWR)
+    funcs_to_check = [μ, bi, bw, a, ξ, γ, μ, γ + μ, k, S, I, W, R]
+    correct = [true for _ in funcs_to_check]
+    inputs = []
+    @test isequal(correct, assess_local_identifiability(de, inputs, funcs_to_check))
```
Comment (ChrisRackauckas) [inline test/mtk_compat.jl:29]: @YingboMa make note that this has a separate way of doing inputs until MTK merges a standard.

**Pull request merged**